app/assets/javascripts/qpixel_api.js

```diff
@@ -151,23 +175,50 @@ window.QPixel = {
     $field.val(prev.substring(0, $field[0].selectionStart) + text + prev.substring($field[0].selectionEnd));
   },
 
+  /**
+   * Used to prevent launching multiple requests to /users/me
+   * @type {Promise<User>|null}
+   */
+  _pendingUser: null,
+
+  /**
+   * @type {User|null}
+   */
   _user: null,
 
   /**
    * Get the user object for the current user.
-   * @returns {Promise<Object>} a JSON object containing user details
+   * @returns {Promise<User>} a JSON object containing user details
    */
   user: async () => {
     if (QPixel._user != null || document.body.dataset.userId === 'none') {
       return QPixel._user;
     }
-    const resp = await fetch('/users/me', {
-      credentials: 'include',
-      headers: {
-        'Accept': 'application/json'
-      }
-    });
-    QPixel._user = await resp.json();
+
+    if(QPixel._pendingUser) {
+        return QPixel._pendingUser
```

Comment:
I think something like this?

```javascript
const myselfPromise = fetch(...).then((v) => {
  QPixel._user = await resp.json();
});
```

Reply:
Hmm, I am not sure what was the reasoning behind me assigning a `myselfPromise` _itself_ to the pending user "locking" check — will take a look and correct if that's an oversight on my part. That does look like one, though!

Reply:
All done — I opted for a bit of a different approach that should work in a similar manner — I added a helper wrapper that always returns a `Response` object and sets one if this is the first one in a sequence of requests to the `/users/me` endpoint:

https://github.com/codidact/qpixel/blob/576602a7d4dcbfd6abff7e2f62cb3ab684b72e54/app/assets/javascripts/qpixel_api.js#L193-L208

There is also a small change in the `user` method that makes sure the body is not re-read multiple times since the pending check is now inlined in the helper (reading a body more than once causes a runtime error as per semantics of `Response` objects):

https://github.com/codidact/qpixel/blob/576602a7d4dcbfd6abff7e2f62cb3ab684b72e54/app/assets/javascripts/qpixel_api.js#L222-L224